lua/session_manager/utils.lua

```diff
@@ -61,6 +61,24 @@ function utils.load_session(filename, discard_current)
     utils.is_session = true
     vim.api.nvim_exec_autocmds('User', { pattern = 'SessionLoadPre' })
     vim.api.nvim_command('silent source ' .. filename)
+    -- close unused lsp clients
+    local bufs = vim.api.nvim_list_bufs()
+    local lsp_clients = vim.lsp.get_active_clients()
+    for _, cli in pairs(lsp_clients) do
+      if cli.config.root_dir ~= vim.fn.getcwd() then
+        local active = false
+        for _, bufnr in pairs(bufs) do
+          for key, _ in pairs(cli.attached_buffers) do
+            if key == bufnr then
+              active = true
+            end
+          end
+        end
+        if active == false then
+          vim.lsp.stop_client(cli.id)
+        end
+      end
+    end
```

Comment:
Some simplifications:
1. Use `vim.loop.cwd()` instead of `vim.fn.cwd()` because `vim.fn.*` functions are slow.
2. Use `vim.tbl_contains` instead of raw loop.
3. Added a break on first buffer match.

```suggestion
    -- Close unused lsp clients
    local bufs = vim.api.nvim_list_bufs()
    local lsp_clients = vim.lsp.get_active_clients()
    for _, client in pairs(lsp_clients) do
      if client.config.root_dir ~= vim.loop.cwd() then
        local active = false
        for _, bufnr in pairs(bufs) do
          if vim.tbl_contains(client.attached_buffers, bufnr) then
            active = true
            break
          end
        end
        if active == false then
          vim.lsp.stop_client(client.id)
        end
      end
    end
```

Reply:
The changes do not work correctly.

Just try:
```
:lua print ( vim.tbl_contains( {[40]=1}, 40 ) )
```

Reply:
`client.attached_buffers` is not a array. Its a dict.

Reply:
Oh, right, sorry. Then instead of `if vim.tbl_contains(client.attached_buffers, bufnr) then` you could use `if client.attached_buffers[bufnr] then`?

Reply:
I don't know if `client.attached_buffers` is possible like this `{ [40] = false, [41]=true }`

Reply:
Oh, okay, makes sense. Let's use `if client.attached_buffers[bufnr] ~= then` just to be sure.

Reply:
Well, perfect change！